src/TBuffer.cpp

```diff
@@ -605,7 +601,43 @@ const QMap<QString, QPair<QString, QVector<QChar>>> TBuffer::csmEncodingTable =
         QChar(0x00E0), QChar(0x00E1), QChar(0x00E2), QChar(0x0103), QChar(0x00E4), QChar(0x00E5), QChar(0x00E6), QChar(0x00E7),  // E0-E7
         QChar(0x00E8), QChar(0x00E9), QChar(0x00EA), QChar(0x00EB), QChar(0x0301), QChar(0x00ED), QChar(0x00EE), QChar(0x00EF),  // E8-EF
         QChar(0x0111), QChar(0x00F1), QChar(0x0323), QChar(0x00F3), QChar(0x00F4), QChar(0x01A1), QChar(0x00F6), QChar(0x00F7),  // F0=F7
-        QChar(0x00F8), QChar(0x00F9), QChar(0x00FA), QChar(0x00FB), QChar(0x00FC), QChar(0x01B0), QChar(0x20AB), QChar(0x00FF)}))}// F8-FF
+        QChar(0x00F8), QChar(0x00F9), QChar(0x00FA), QChar(0x00FB), QChar(0x00FC), QChar(0x01B0), QChar(0x20AB), QChar(0x00FF)})}// F8-FF
+};
+
+QMap<QString, QString> TBuffer::smEncodingNamesMap  = {
+    {QStringLiteral("ASCII"), tr("ASCII (Telnet default encoding)", "Server encoding name map: Ensure both instances have same translation text (1 of 2)")},
```

Comment:
I don't know what do you mean here.

Reply:
The `Developer's comment` ? I have found it is advisable to have something like this to ensure that BOTH occurrences of `ASCII (Telnet default encoding)` get modified in step - if this instance of this source string gets modified it highlights that there is an instance in this or another class that ALSO must be both edited and translated as well.  By default if there was not such a comment then only the first case of `ASCII (Telnet default encoding)` gets displayed in `QtLinguist` by default and if ONLY this instance in the source code gets edited then there can be programme breakage when the other instance gets used.  This is the issue when using `QStrings` as identifiers internally - and why `QFlags`/`enums` are a better solution (c.f. custom line styles...)

Reply:
Ah ok, how about saying "ensure this text is translated the same"?

Reply:
- need to keep the X of Y part but in principle I guess I can shorten it - and invalidate all the current translations now using it... 🙁

Reply:
Sorry, but these kinds of adjustments would happen given they were made before the PR was even up. Isn't it possible to just find & replace in a file?

Reply:
Well there is a multiplication factor i.e. change anything in a `tr(`...`)` and it is not just that source file but every (well apart from the _plurals only_ `mudlet_en_US.ts`) file changes as well - before I pushed out the PR I got to do it all behind the scenes; however going forward coders ought to at be considering _"do I really need to change this QString?"_

TBH I suspect we will have to re-examine a lot of the textual content in Mudlet and once this PR goes in I think a couple of multi-lingual Mudlet developers will need to go through and do a translation in, say German and Russian 😉 just so we can identify stuff that is not flagged for translation but which should be.  This will likely require us to rethink some code fragments (other languages strings will not necessary be the same size on screen which may break some dialogs) and I suspect it will better all around if we do this for a couple of languages before we open it up for collaborative Web translation so we are not asking people to work on strings that have to be thrown away...

 At some stage we will want to include the defines for `QT_NO_CAST_FROM_ASCII` and `QT_NO_CAST_TO_ASCII` in the CMake/QMake project files but we can do that on a class by class basis at the moment.

Reply:
I don't understand why can't we revise the translation hint text to make sense to translators.

You have to understand not everybody is a technical coder who understands what terms like "instance" will mean.

Reply:
Fair enough - I'll give it a tweak...

Reply:
Thank you 👼

Reply:
☑️ Have changed all _instance_ to _case_ so that there is a fairly standardised: 

> This text is used programmatically, ensure all cases have same translation (x of y)

normally this is to signal that the C++ code relies on the same (translatable) text to be used at different points in the codebase and should only be encountered in dialogues which are only around for a short time and it is not worthwhile to try and code to handle the text being changed by a GUI language change happening between where one ~instance~case of the string is used and another.

In this case _that_ is not the case, instead it is because the use is in two separate classes (`TBuffer` and `dlgProfilePreferences`) and it may not be obvious that they are linked and a developer changing one ~instance~case needs to remember to change the other as well; so the text is now:

> Server encoding name map: ensure all cases have same translation text (x of y)

😀